code/create_survey_geojson.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+import json
+import re
+from pathlib import Path
+from typing import Iterable, Optional
+
+import pandas as pd
+
+# path to data directory
+DPATH_DATA = Path(__file__).parent.parent / "data"
+
+# path to assets directory (for GeoJSON data input/output)
+DPATH_ASSETS = Path(__file__).parent / "assets"
+
+# path to generic US states GeoJSON file
+FPATH_STATES_JSON_DEFAULT = DPATH_ASSETS / "us_states.json"
+
+# path to output GeoJSON file
+FPATH_OUT_DEFAULT = DPATH_ASSETS / "survey_states.json"
+
+
+def create_survey_geojson(
+    survey_states_and_clusters: Iterable[str],
+    fpath_states_json: Path | str = FPATH_STATES_JSON_DEFAULT,
+    fpath_out: Optional[Path | str] = FPATH_OUT_DEFAULT,
+) -> dict:
+    """Create a GeoJSON file for the states and state clusters used in the survey.
+
+    Parameters
+    ----------
+    survey_states_and_clusters : Iterable[str]
+        State and state cluster names to include in the GeoJSON file. State
+        cluster names should be in the format "State1, State2, ... (Cluster X)".
+    fpath_states_json : Path | str, optional
+        Path to GeoJSON file with standard US states.
+        By default, will use the one that was
+        downloaded from https://github.com/PublicaMundi/MappingAPI/blob/master/data/geojson/us-states.json.
```

Comment:
cool find!! just curious: from a quick google, it looks like there's a few different options for US states geojson files (not sure how different they are) - how did you land on this specific resource? :)

Reply:
I think it was the first one that I saw, and it was identical to the one linked [here](https://community.plotly.com/t/data-files-from-geojson-marker-clustering-choropleth-map-etc-examples/54867), though I am not sure if Dash Leaflet is actually related to Plotly Dash or not. 

Initially I wanted to use the one used by Plotly by default, but couldn't find it. It turns out [they use a format other than GeoJSON](https://github.com/plotly/plotly.py/tree/c6043797c15ab96450aaa2e6becb20c534672ab4/packages/python/plotly-geo/_plotly_geo/package_data), and at that point I had already written the GeoJSON code and didn't really want to change it. Also the default Plotly map has holes in Florida and Utah which I think are large bodies of water, and I personally am not a big fan of those.